src/test/operators/table_scan_test.cpp

```diff
@@ -271,6 +273,38 @@ TEST_P(OperatorsTableScanTest, SingleScanReturnsCorrectRowCount) {
   EXPECT_TABLE_EQ_UNORDERED(scan->get_output(), expected_result);
 }
 
+TEST_P(OperatorsTableScanTest, SingleScanWithSubselect) {
+  std::shared_ptr<Table> expected_result = load_table("resources/test_data/tbl/int_float_filtered2.tbl", 1);
+
+  const auto pqp =
+      std::make_shared<Limit>(std::make_shared<Projection>(get_int_string_op(), expression_vector(to_expression(1234))),
+                              to_expression(int64_t{1}));
+
+  auto scan = std::make_shared<TableScan>(get_int_float_op(),
+                                          greater_than_equals_(pqp_column_(ColumnID{0}, DataType::Int, false, "a"),
+                                                               pqp_select_(pqp, DataType::Int, false)));
+  scan->execute();
+  EXPECT_EQ(scan->impl_description(), "ColumnVsValue");
+  EXPECT_TABLE_EQ_UNORDERED(scan->get_output(), expected_result);
+}
+
+TEST_P(OperatorsTableScanTest, BetweenScanWithSubselect) {
+  std::shared_ptr<Table> expected_result = load_table("resources/test_data/tbl/int_float_filtered2.tbl", 1);
+
+  const auto pqp =
```

Comment:
a name like "subselect_pqp" would immediately give away what this is, would be a minor improvement over the two seconds it took me to understand this with the current name.